runtime-sdk/modules/evm/src/lib.rs

```diff
@@ -214,47 +214,10 @@ impl From<evm::ExitFatal> for Error {
 fn process_evm_result(exit_reason: evm::ExitReason, data: Vec<u8>) -> Result<Vec<u8>, Error> {
     match exit_reason {
         evm::ExitReason::Succeed(_) => Ok(data),
-        evm::ExitReason::Revert(_) => {
-            if data.is_empty() {
-                return Err(Error::Reverted("no revert reason".to_string()));
-            }
-
-            // Decode revert reason, format is as follows:
-            //
-            // 08c379a0                                                         <- Function selector
-            // 0000000000000000000000000000000000000000000000000000000000000020 <- Offset of string return value
-            // 0000000000000000000000000000000000000000000000000000000000000047 <- Length of string return value (the revert reason)
-            // 6d7946756e6374696f6e206f6e6c79206163636570747320617267756d656e74 <- First 32 bytes of the revert reason
-            // 7320776869636820617265206772656174686572207468616e206f7220657175 <- Next 32 bytes of the revert reason
-            // 616c20746f203500000000000000000000000000000000000000000000000000 <- Last 7 bytes of the revert reason
-            //
-            const ERROR_STRING_SELECTOR: &[u8] = &[0x08, 0xc3, 0x79, 0xa0]; // Keccak256("Error(string)")
-            const FIELD_OFFSET_START: usize = 4;
-            const FIELD_LENGTH_START: usize = FIELD_OFFSET_START + 32;
-            const FIELD_REASON_START: usize = FIELD_LENGTH_START + 32;
-            const MIN_SIZE: usize = FIELD_REASON_START;
-            const MAX_REASON_SIZE: usize = 1024;
-
-            let max_raw_len = data.len().clamp(0, MAX_REASON_SIZE);
-            if data.len() < MIN_SIZE || !data.starts_with(ERROR_STRING_SELECTOR) {
-                return Err(Error::Reverted(base64::encode(&data[..max_raw_len])));
-            }
-            // Decode and validate length.
-            let mut length =
-                primitive_types::U256::from(&data[FIELD_LENGTH_START..FIELD_LENGTH_START + 32])
-                    .low_u32() as usize;
-            if FIELD_REASON_START + length > data.len() {
-                return Err(Error::Reverted(base64::encode(&data[..max_raw_len])));
-            }
-            // Make sure that this doesn't ever return huge reason values as this is at least
-            // somewhat contract-controlled.
-            if length > MAX_REASON_SIZE {
-                length = MAX_REASON_SIZE;
-            }
-            let reason =
-                String::from_utf8_lossy(&data[FIELD_REASON_START..FIELD_REASON_START + length]);
-            Err(Error::Reverted(reason.to_string()))
-        }
+        // Encode raw revert reason as Base64, up to the maximum length of 1024 bytes.
+        evm::ExitReason::Revert(_) => Err(Error::Reverted(base64::encode(
+            &data[..data.len().clamp(0, 1024)],
```

Comment:
I expect it to be unlikely that there will be 1kb of errors returned via revert.  After base64 encoding the full return value will be approx 1400 bytes.

Is there a specific reason that reverts are limited to 1kb but success isn't?

Reply:
Good point, let me think about it.